src/integration/vliw-integration.ts

```diff
@@ -277,15 +279,26 @@ export class VLIWIntegration extends MachineIntegration {
         if (!this.stopCondition) {
             setTimeout(() => {
                 let machineStatus = this.stepForward();
-                if (!(machineStatus === VLIWError.BREAKPOINT || machineStatus === VLIWError.ENDEXE)) {
-                    this.executionLoop(speed);
-                } else {
-                    if (machineStatus === VLIWError.BREAKPOINT) {
+                let stop: boolean = true;
+                switch (machineStatus) {
+                    case VLIWError.OK:
+                    case VLIWError.PCOUTOFRANGE: //TODO: is this really an error? We always go out of range when we finish the execution or there is a branch at the end
+                        stop = false;
+                        break;
+                    case VLIWError.BREAKPOINT:
                         alert(t('execution.stopped'));
-                    } else if (machineStatus === VLIWError.ENDEXE) {
+                        break;
+                    case VLIWError.ENDEXE:
                         this.finishedExecution = true;
                         alert(t('execution.finished'));
-                    }
+                        break;
+                    default:
+                        alert(t('execution.error') + ": " + VLIWError[machineStatus]);
```

Comment:
These alerts and the successful execution ones above are also good candidates for the alert component implementation in the future